config/src/test/scala/com/typesafe/config/impl/ConcatenationTest.scala

```diff
@@ -345,26 +345,19 @@ class ConcatenationTest extends TestUtils {
         assertEquals(Seq(1, 2, 3), conf.getObjectList("x.a").asScala.toList.map(_.toConfig.getInt("b")))
     }
 
-    // We would ideally make this case NOT throw an exception but we need to do some work
-    // to get there, see https://github.com/typesafehub/config/issues/160
     @Test
     def plusEqualsMultipleTimesNestedInArray() {
-        val e = intercept[ConfigException.Parse] {
-            val conf = parseConfig("""x = [ { a += 1, a += 2, a += 3 } ] """).resolve()
-            assertEquals(Seq(1, 2, 3), conf.getObjectList("x").asScala.toVector(0).toConfig.getIntList("a").asScala.toList)
-        }
-        assertTrue(e.getMessage.contains("limitation"))
+        val conf = parseConfig("""x = [ { a += 1, a += 2, a += 3 } ] """).resolve()
+        assertEquals(Seq(1, 2, 3), conf.getObjectList("x").asScala.toVector(0).toConfig.getIntList("a").asScala.toList)
     }
 
-    // We would ideally make this case NOT throw an exception but we need to do some work
-    // to get there, see https://github.com/typesafehub/config/issues/160
     @Test
     def plusEqualsMultipleTimesNestedInPlusEquals() {
-        val e = intercept[ConfigException.Parse] {
+        val e = intercept[ConfigException.BugOrBroken] {
```

Comment:
If we want this to be invalid syntax, we should be throwing the Parse exception or a resolve-related exception, not the BugOrBroken.

I don't understand why it's invalid though; it expands to something like this right:

```
 x = ${?x} [ { a = ${?a} [1], a = ${?a} [2], a = ${?a} [3] } ]
```

It seems logical to me - the only "twist" here is that we have two nested lists, right? Or am I missing something?

The resolve code is definitely "fun" but it's probably possible to fix this.